database.go

```diff
@@ -140,6 +141,19 @@ func (db *Database) LockProtectedEntries() error {
 	return nil
 }
 
+func (db *Database) GetBinaries() *Binaries {
```

Comment:
Does this method need to be exposed?

A good interface would be where you do not really have to fuzz with the binaries directly. `AddBinary` already makes it unnecessary to have this method exposed for adding binaries.

The cleanup part will remove the need to remove binaries from the slice manually.

Reply:
Ideally it should not, but in my client I use direct access to the database from several places.

Reply:
For example I would like to have screen of all the binaries in my app. Name, size, content and which entries linked to particular binary.

Reply:
Alternatively I can build map entry to binary of course. But in that case I'll not find unlinked binaries and can not notify user about possibility to clean it. This is very edge case, I think we can remove direct access ti binaries anyway and then provide some alternative if needed.

Reply:
The user should not have to decide whether they clean up or not. Most users will not understand what the implications are unless you clearly outline it.

If the user removes a file from an entry, the intention would be clear to me: I do not need this file anymore.

The structure would remove the binary reference directly.
That means we could theoretically remove the binary from the list then and there. The main reason not to do it there is because it might be slow and because it is not clear whether the file is referenced elsewhere.

Theoretically, editing an entry would mean duplicating that entry, adding it to the entry history and only removing the binary reference from the latest version.

In that case binaries would not be removed later either unless the option to clean up means truncating history.